L/LibCURL/common.jl

```diff
@@ -0,0 +1,107 @@
+
+    if name == "CURL"
+        this_is_curl_jll = true
+    elseif name == "LibCURL"
+        this_is_curl_jll = false
+    else
+        msg = "Not a valid name: $(name). Valid names are: LibCURL, CURL"
+        throw(ArgumentError(msg))
+    end
+
+    # Collection of sources required to build LibCURL
+    sources = [
+        ArchiveSource("https://curl.se/download/curl-$(version).tar.gz", hash),
+    ]
+
+    # Bash recipe for building across all platforms
+    script = raw"""
+    cd $WORKSPACE/srcdir/curl-*
+
+    # Holy crow we really configure the bitlets out of this thing
+    FLAGS=(
+        # Disable....almost everything
+        --without-ssl --without-gnutls --without-gssapi
+        --without-libidn --without-libidn2 --without-librtmp
+        --without-nss --without-polarssl
+        --without-spnego --without-libpsl --disable-ares --disable-manual
+        --disable-ldap --disable-ldaps --without-zsh-functions-dir
+        --disable-static --without-libgsasl
+
+        # A few things we actually enable
+        --with-libssh2=${prefix} --with-zlib=${prefix} --with-nghttp2=${prefix}
+        --enable-versioned-symbols
+    )
+
+
+    if [[ ${target} == *mingw* ]]; then
+        # We need to tell it where to find libssh2 on windows
+        FLAGS+=(LDFLAGS="${LDFLAGS} -L${prefix}/bin")
+
+        # We also need to tell it to link against schannel (native TLS library)
+        FLAGS+=(--with-schannel)
+    elif [[ ${target} == *darwin* ]]; then
+        # On Darwin, we need to use SecureTransport (native TLS library)
+        FLAGS+=(--with-secure-transport)
+
+        # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+        # complain about: `Symbol not found: ___isOSVersionAtLeast`
+        if [[ "${target}" == aarch64* ]]; then
+            export CFLAGS=-mmacosx-version-min=11.0
+        else
+            export CFLAGS=-mmacosx-version-min=10.11
+        fi
+    else
+        # On all other systems, we use MbedTLS
+        FLAGS+=(--with-mbedtls=${prefix})
+    fi
+
+    ./configure --prefix=$prefix --host=$target --build=${MACHTYPE} "${FLAGS[@]}"
+    make -j${nproc}
+    make install
+    install_license COPYING
+    """
+
+    # These are the platforms we will build for by default, unless further
+    # platforms are passed in on the command line
+    platforms = supported_platforms(;experimental=true)
+
+    # The products that we will ensure are always built
+    if this_is_curl_jll
+        # The Curl_jll package only provides the executable
+        products = [
+            ExecutableProduct("curl", :curl),
+        ]
+    else
+        # The LibCURL_jll *package* provides both the library and the executable.
+        # However, the LibCURL_jll *stdlib* only provides the library.
+        products = [
+            LibraryProduct("libcurl", :libcurl),
+            ExecutableProduct("curl", :curl),
+        ]
```

Comment:
Since the stdlib exposes only libcurl, I'm not sure this should expose both of them, it'll only confuse things.

Reply:
I've removed the executable product from the LibCURL_jll package.

Reply:
> I just copied what is currently done on Ygg master:

I didn't realise that... sigh.  So these means they have conflicting files, which is not good.  I presume there is a way to build only libcurl?  And perhaps curl using libcurl from the prefix.  At the moment we're still producing two identical tarballs, which is a waste.

Reply:
You mean https://github.com/JuliaLang/julia/blob/0b48b91c9881823d860eaf0a605072b7f20a4cbb/stdlib/LibCURL_jll/src/LibCURL_jll.jl?

Reply:
Yeah.

Reply:
As far as I know, all those files have been created manually.  I don't think there is any "code that generates" them.

Reply:
And also, where is the code that makes it so that the `LibCURL_jll` stdlib JLL doesn't include the executable product, even though the `LibCUL_jll` package[^1] does include the executable product.

[^1]: https://github.com/JuliaBinaryWrappers/LibCURL_jll.jl

Reply:
> As far as I know, all those files have been created manually. I don't think there is any "code that generates" them.

Oh, I see.

Reply:
(side note: I'm kind of curious why you don't just use hyperlinks instead of having footnotes, which requires _more_ clicks on long posts :slightly_smiling_face:)

Reply:
😂 😂 😂

Reply:
No particular reason at all. I suppose I just picked up the habit at one point 😂